src/ImageSharp/Processing/Processors/Convolution/KernelSamplingMap.cs

```diff
@@ -99,8 +99,8 @@ public void Dispose()
         {
             if (!this.isDisposed)
             {
-                this.yOffsets.Dispose();
-                this.xOffsets.Dispose();
+                this.yOffsets?.Dispose();
```

Comment:
Are you failing to call `BuildSamplingOffsetMap` at some point?

Reply:
Yes, in that `WorksWithDiscoBuffers` test. One of the `Allocate` calls in the `BuildOffsetsMap` call was throwing due to insufficient memory, so `Dispose` was called, and then it was throwing here because the other buffer was still `null`, as it crashed before that was assigned 🤔

Reply:
You running in 32bit? How big are the buffers?

Reply:
The one crashing is for `xOffsets`, which is `100` in width and `65` in kernel size, so `6500` in total.
I'm running in 64-bit by default (double checked and `sizeof(IntPtr) == 8`).

Reply:
Hmmm... That should be nothing. I'll pull it down tomorrow and see what is happening.

Reply:
``` 
Requested allocation: 19500 elements of Int32 is over the capacity of the MemoryAllocator
```

Reply:
That number is given by bounds.Height of 300 x kernel.Height of 65. Is that large a kernel expected?

Reply:
So... You're actually requesting 78000 bytes during the request which is not an unreasonable amount. I think that test limit is simply set far to low. 

The available allocation is calculated as follows:

`no. of rows * width * sizeof(pixel)`

Now since the test is using a `300 * 100 * Bgr24` with a limit of 41 rows this means the maximum allowable buffer is. 

`41 * 100 * 3 = 12300` bytes which is far too low. We need 260 rows which still allows for discontiguous testing.

Reply:
> Is that large a kernel expected?

The default radius is 32, so yeah that matches the expected size for the default parameters 🙂